packages/salesforcedx-vscode-apex/src/commands/forceGenerateFauxClasses.ts

```diff
@@ -190,6 +190,15 @@ export async function forceGenerateFauxClassesCreate(
 export async function initSObjectDefinitions(projectPath: string) {
   const hasDefaultUsernameSet =
```

Comment:
Isn't the root of the problem in this evaluation ? In the past we were already checking for this and the projectPath before triggering the refresh.

Reply:
Yes, this is the root of the problem. We still want to throw the error as usual when someone has no authorized org and runs the "Refresh SObjects" manually. But, when we prompt them with an option to refresh sobjects, we make sure org is authorized and hence call initSObjectDefinitionsWithUsername.
Ideally I hate this name and would prefer simple function overloading, but typescript doesn't allow it.

Thinking about it now, I think the better naming will be 
initSObjectDefinitions(projectPath: string) changed to verifyUsernameAndInitSObjectDefinitions(projectPath: string) -> Used by the command
initSObjectDefinitionsWithUserName(projectPath: string, hasDefaultUsernameSet: boolean) changed to initSObjectDefinitions(projectPath: string) -> user by our new refresh button and we have made sure username has been set.